ha/util/actuator_req_manager.py

```diff
@@ -0,0 +1,237 @@
+
+        Return:
+            created actuator request
+        """
+
+        with open(ACTUATOR_SCHEMA, 'r') as actuator_req_schema_file:
+            actuator_req = json.load(actuator_req_schema_file)
+
+        # Get details for current node
+        node_name = Conf.get(const.HA_GLOBAL_INDEX, f"CLUSTER_MANAGER{_DELIM}local_node")
+
+        key_val  = self._conf_store.get(f"{PVTFQDN_TO_NODEID_KEY}/{node_name}")
+        _, node_id = key_val.popitem()
+
+        system_health = SystemHealth(self._conf_store)
+        health_manager = SystemHealthManager(self._conf_store)
+        key = system_health._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)
+
+        node_map_val = health_manager.get_key(key)
+        if node_map_val is None:
+            raise ClusterManagerError("Failed to fetch node_map value")
+
+        node_map_dict = ast.literal_eval(node_map_val)
+        self._generate_uuid()
+
+        # Populate the actuator request schema
+        actuator_req[ACTUATOR_ATTRIBUTES.TIME] = str(int(time.time()))
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.SITE_ID] = node_map_dict[NODE_MAP_ATTRIBUTES.SITE_ID.value]
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.RACK_ID] = node_map_dict[NODE_MAP_ATTRIBUTES.RACK_ID.value]
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.REQUEST_PATH][ACTUATOR_ATTRIBUTES.NODE_ID] = node_id
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.TARGET_NODE_ID] = target_node_name
+        actuator_req[ACTUATOR_ATTRIBUTES.MESSAGE][ACTUATOR_ATTRIBUTES.HEADER][ACTUATOR_ATTRIBUTES.UUID] = self._uuid
+
+        actuator_req_schema_file.close()
+        req = json.dumps(actuator_req)
+
+        return req
+
+    def _send_req(self, req):
+        """
+        Send the created request to "monitor" on message bus
+
+        Args:
+            req : created actuator request
+        """
+
+        self.req_message_type = Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}req_message_type")
+        self.producer_id = Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}producer_id")
+        self.producer =  MessageBus.get_producer(self.producer_id, self.req_message_type)
+
+        Log.debug(f"Publishing request {req} on message_type {self.req_message_type}")
+        self.producer.publish(req)
+
+    def _wait_for_resp(self):
+        """
+        Register to wait for a response to the sent request.
+        """
+        self.consumer_group =  Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}consumer_group")
+        self.consumer_id =  Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}consumer_id")
+        self.resp_message_type = Conf.get(const.HA_GLOBAL_INDEX, f"ACTUATOR_MANAGER{_DELIM}resp_message_type")
+
+        self.consumer = MessageBus.get_consumer(consumer_id=str(self.consumer_id),
```

Comment:
Consumer should be created with "latest" offset which also means create consumer before sending the request.

Reply:
Done

Reply:
By default offset is "earliest". offset = "latest" must be passed to get_consumer ?

Reply:
I see this too.